src/webserver/DOSGuard.h

```diff
@@ -40,22 +46,26 @@ class DOSGuard
     {
         using T = std::unordered_set<std::string> const;
 
-        auto const& c = getConfig(config);
-        if (!c)
-            return T();
-
-        if (!c->contains("whitelist"))
-            return T();
+        try
+        {
+            auto const& c = getConfig(config);
+            if (!c)
+                return T();
 
-        auto const& w = c->at("whitelist").as_array();
+            auto const& w = c->at("whitelist").as_array();
 
-        auto const transform = [](auto const& elem) {
-            return std::string(elem.as_string().c_str());
-        };
+            auto const transform = [](auto const& elem) {
+                return std::string(elem.as_string().c_str());
+            };
 
-        return T(
-            boost::transform_iterator(w.begin(), transform),
-            boost::transform_iterator(w.end(), transform));
+            return T(
+                boost::transform_iterator(w.begin(), transform),
+                boost::transform_iterator(w.end(), transform));
```

Comment:
I know this is not actually part of this PR, but can we do this with the standard library instead of boost?

Reply:
It depends on what you mean by that. There's no `transform_iterator` in the standard library.

Reply:
I don't think there's a concise way to do this without using boost.